addons/project/models/project_task.py

```diff
@@ -1590,3 +1595,16 @@ def action_redirect_to_project_task_form(self):
             'url': '/web#model=project.task&id=%s&action=%s&view_type=form' % (self.id, self.env.ref('project.action_view_my_task').id),
             'target': 'new',
         }
+
+    @api.model
+    def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=False, lazy=True): #TODO: multiple groupby ??
+        # A read_group can not be performed if records are grouped by personal_stage_type_id as it is a computed field.
+        # personal_stage_type_ids behaves like a M2O from the point of view of the user, we therefore use this field instead.
+        if 'personal_stage_type_id' in groupby and (not lazy or groupby[0] == 'personal_stage_type_id'):
+            groupby = list(map(lambda field: "personal_stage_type_ids" if field == "personal_stage_type_id" else field, groupby)) # limitation, problem when grouped on personal_stage_type_id and personal_stage_type_ids but who does that?
+            result = super().read_group(domain, fields, groupby, offset, limit, orderby, lazy)
+            for group in result:
+                group['personal_stage_type_id'] = group.pop('personal_stage_type_ids', False)
+                group['personal_stage_type_id_count'] = group.pop('personal_stage_type_ids_count', False)
```

Comment:
Why do we pop those? :thinking:

Reply:
> Why do we pop those? 🤔

We want to replace occurrences of personal_stage_type_ids (M2M) in each group by personal_stage_type_id (M2O). So I use pop in order to remove the M2M from the group while copying its value.